apps/system/js/action_menu.js

```diff
@@ -1,326 +1,175 @@
+  var ActionMenu = function(controller) {
+    this.instanceID = 'action-menu';
+    this.controller = controller;
+    this.onselected = controller.successCb || function() {};
+    this.oncancel = controller.cancelCb || function() {};
+    this.options = {};
 
     /**
-     * Whether or not the ActionMenu is visible.
-     * @memberof ActionMenu.prototype
-     * @return {Boolean} The ActionMenu is visible.
+     * render the dialog
      */
-    get visible() {
-      return this.container.classList.contains('visible');
-    },
-
-    /**
-     * Builds dom and adds event listeners
-     * @memberof ActionMenu.prototype
-     */
-    start: function() {
-      // Create the structure
-      this.container = document.createElement('form');
-      this.container.dataset.type = 'action';
-      this.container.setAttribute('role', 'dialog');
-      this.container.setAttribute('data-z-index-level', 'action-menu');
-
-      // An action menu has a mandatory header
-      this.header = document.createElement('header');
-      if (this.titleL10nId !== undefined) {
-        this.header.setAttribute('data-l10n-id', this.titleL10nId);
-      }
-
-      this.container.appendChild(this.header);
-
-      // And a default choice button if asked
-      if (this.askForDefaultChoice) {
-        this.defaultChoice = document.createElement('label');
-        this.defaultChoice.setAttribute('class', 'pack-checkbox');
-        this.defaultChoice.setAttribute('data-action', 'set-default-action');
-
-        this.defaultChoiceInput = document.createElement('input');
-        this.defaultChoiceInput.setAttribute('type', 'checkbox');
-
-        this.defaultChoiceSpan = document.createElement('span');
-        this.defaultChoiceSpan.setAttribute('data-l10n-id',
-                                            'set-default-action');
-
-        this.defaultChoice.appendChild(this.defaultChoiceInput);
-        this.defaultChoice.appendChild(this.defaultChoiceSpan);
-      }
+    this.render();
+    this.publish('created');
+  };
 
-      // Following our paradigm we need a cancel
-      this.cancel = document.createElement('button');
-      this.cancel.setAttribute('data-l10n-id', 'cancel');
-      this.cancel.dataset.action = 'cancel';
+  ActionMenu.prototype = Object.create(SystemDialog.prototype);
 
-      // We have a menu with all the options
-      this.menu = document.createElement('menu');
-      this.container.appendChild(this.menu);
+  ActionMenu.prototype.customID = 'action-menu';
 
-      // We append to System app (actually to '#screen')
-      var screen = document.getElementById('screen');
-      screen.appendChild(this.container);
-      screen.classList.add('action-menu');
+  ActionMenu.prototype.DEBUG = false;
 
-      this.buildMenu(this.listItems);
+  ActionMenu.prototype.name = 'ActionMenu';
 
-      this.container.addEventListener('submit', this);
-      this.menu.addEventListener('click', this);
+  ActionMenu.prototype.EVENT_PREFIX = 'actionmenu';
```

Comment:
just a thought: I am not sure if we need to have a special prefix for action menu. It would be easy to manage if all system dialogs share the same prefix (I forgot the reason that SimLock had its own prefix). But let's see if we have more system dialogs in the future, what will happen.